src/Raven.Studio/typescript/viewmodels/manage/certificates.ts

```diff
@@ -126,26 +162,57 @@ class certificates extends viewModelBase {
     
     private filterCertificates() {
         const filter = this.nameFilter().toLocaleLowerCase();
-        const clearance = this.clearanceFilter();
         
-        this.certificates().forEach(certificate => {
+        this.certificates().forEach((certificate: unifiedCertificateDefinitionWithCache) => {
+            
             const nameMatch = !certificate || certificate.Name.toLocaleLowerCase().includes(filter);
-            const clearanceMatch = !clearance || certificate.SecurityClearance === clearance;
             const thumbprintMatch = !certificate || _.some(certificate.Thumbprints, x => x.toLocaleLowerCase().includes(filter));
-            certificate.Visible((nameMatch || thumbprintMatch) && clearanceMatch);
+
+            const validMatch = !certificate || (this.showValidCertificates() && !certificate.expirationClass);
```

Comment:
those this logic to function and return if there is no match. No need to calculate all those things

Reply:
done